charts/cht-chart-4x/templates/api-deployment.yaml

```diff
@@ -33,6 +33,7 @@ spec:
             - name: API_PORT
               value: '5988'
           image: {{ .Values.upstream_servers.docker_registry }}/cht-api:{{ .Values.cht_image_tag }}
+          imagePullPolicy: Always
```

Comment:
Instead of setting the value here directly, we should add a variable in values and pull from there.
Something like adding to the values.yaml
`imagePullPolicy: IfNotPresent  # default value`
The reason for this is that for production deployments images should have an immutable tag with the cluster pulling only if it's not present. Otherwise, it causes issues like increasing egress costs and network activity/load on the cluster. `Always` is okay for CI/CD and for dev or debugging deployments.

Reply:
I'm afraid we will run into a situation where we will need this in production too, where some prod instance is on a branch build because they needed some urgent hotpatch, and we can't upgrade them. 
We don't do upgrades often enough to have frequent image pulls?

Reply:
>I'm also reluctant to be so opinionated in the values file. Someone would need to have knowledge about what the consequences are for changing this value

Folks could even leave out the key from the values.yaml and helm will apply the defaults you originally build it with in this PR.

>I'm afraid we will run into a situation where we will need this in production too, where some prod instance is on a branch build because they needed some urgent hotpatch, and we can't upgrade them.

If we need to make an urgent patch to the production deployment, these should be rare and one-off and therefore editing the deployment object in k8s is the way to go.

We have things in the values.yml that we expect to change rarely. This setting could sit next to those. I think the main thing here though is that for production deployments we need to stick to the principle that images should have an immutable tag.

Reply:
Otherwise, we can't have deterministic deployments in production and we'd have to introduce [image digests](https://cloud.google.com/kubernetes-engine/docs/concepts/about-container-images) which I believe would be an overkill and very much like to avoid.

Reply:
> deterministic deployments

When would the images be pulled, in normal operations?

Reply:
If the policy is set to `Always`, that could happen if the pod restarts for whatever reason. Not just at the time of install.